packages/dd-trace/src/debugger/index.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+const { join } = require('path')
+const { Worker, MessageChannel } = require('worker_threads')
+const log = require('../log')
+
+let worker = null
+let configChannel = null
+
+module.exports = {
+  start,
+  configure
+}
+
+function start (config, rc) {
+  if (worker !== null) return
+
+  log.debug('Starting Dynamic Instrumentation client...')
+
+  const rcAckCallbacks = new Map()
+  const rcChannel = new MessageChannel()
+  configChannel = new MessageChannel()
+
+  rc.setProductHandler('LIVE_DEBUGGING', (action, conf, id, ack) => {
```

Comment:
going back to the race condition - I see that the postMessage happen without waiting for the previous one to complete. is that right?

Reply:
> going back to the race condition

Which race condition? I think I've mentioned several. But to my knowledge I've dealt with all of them, so there should be none left that I know of.

> I see that the postMessage happen without waiting for the previous one to complete. is that right?

That's correct. I don't think there's an issue with that. Have you spotted anything that I don't see?

Reply:
We had a chat about this over Zoom and I now understand the race condition you're referring to:

If, say, there's just a single breakpoint active and an RC request comes in to remove that breakpoint, after removing it, we post `Debugger.disable` on the inspector protocol session since there's no reason to keep the debugger session active if there's no active breakpoints.

However, if while we process the request to remove the last breakpoint, but before disabling the inspector session, get a new RC request to add a new breakpoint, we will not post `Debugger.enable` because we see that the session is already active.

This is the race condition.

Reply:
For now I added a mutex lock. I'll revet that change if we decided to just never tear it down